shared/p2p/service.go

```diff
@@ -57,22 +58,28 @@ func NewServer() (*Server, error) {
 	}
 
 	return &Server{
-		ctx:          ctx,
-		cancel:       cancel,
-		feeds:        make(map[reflect.Type]Feed),
-		host:         h,
-		gsub:         gsub,
-		mutex:        &sync.Mutex{},
-		topicMapping: make(map[reflect.Type]string),
+		ctx:           ctx,
+		cancel:        cancel,
+		feeds:         make(map[reflect.Type]Feed),
+		host:          h,
+		gsub:          gsub,
+		mutex:         &sync.Mutex{},
+		topicMapping:  make(map[reflect.Type]string),
+		bootstrapNode: bootstrapNode,
 	}, nil
 }
 
 // Start the main routine for an p2p server.
 func (s *Server) Start() {
 	log.Info("Starting service")
-	if err := startDiscovery(s.ctx, s.host); err != nil {
-		log.Errorf("Could not start p2p discovery! %v", err)
-		return
+
+	if err := startDHTDiscovery(s.ctx, s.host, s.bootstrapNode); err != nil {
```

Comment:
@prestonvanloon Let me know if this makes sense...

Node first tries to do DHT discovery, if fails then it tries to do mDNS discovery